packages/spacecat-shared-utils/src/aem.js

```diff
@@ -203,20 +205,49 @@ export function detectAEMVersion(htmlSource, headers = {}) {
 /**
  * Determines the AEM CS page ID for Content API, from the page URL
  * @param {string} pageURL - The URL of the page
+ * @param {string} authorURL - The URL of the author instance
+ * @param {string} bearerToken - The access token for the page
  * @return {string|null} - The AEM CS page ID
  */
-export async function determineAEMCSPageId(pageURL) {
-  const htmlResponse = await fetch(pageURL);
-  if (!htmlResponse.ok) {
-    return null;
-  }
-  const html = await htmlResponse.text();
-  const metaTagRegex = /<meta\s+name=['"]content-page-id['"]\s+content=['"]([^'"]*)['"]\s*\/?>/i;
-  const match = html.match(metaTagRegex);
-
-  let pageId = null;
-  if (match && match[1] && match[1].trim() !== '') {
-    pageId = match[1].trim();
+export async function determineAEMCSPageId(pageURL, authorURL, bearerToken) {
+  try {
+    const htmlResponse = await fetch(pageURL);
+    if (!htmlResponse.ok) {
+      return null;
+    }
+    const html = await htmlResponse.text();
+    let pageId = null;
+    if (authorURL && bearerToken) {
+      // try content-page-ref
+      const contentPageRefRegex = /<meta\s+name=['"]content-page-ref['"]\s+content=['"]([^'"]*)['"]\s*\/?>/i;
+      const contentPageRefMatch = html.match(contentPageRefRegex);
+      let contentPageRef = null;
+      if (contentPageRefMatch?.[1]?.trim()) {
+        contentPageRef = contentPageRefMatch[1].trim();
+        const referenceResolveResponse = await fetch(
+          `${authorURL}${CONTENT_API_PAGES_RESOLVE_ENDPOINT}?pageRef=${contentPageRef}`,
```

Comment:
This is not supported in AMS yet. @terinmez I thought we had agreed to send the contentPageRef directly to PSS, since the PSS bundle can handle it. Could you clarify if that’s still the plan?